mail/src/main/java/org/jboss/as/mail/extension/MailSessionDefinition.java

```diff
@@ -89,6 +84,19 @@ private MailSessionDefinition() {
 
     );
 
+    static final MailSessionDefinition INSTANCE = new MailSessionDefinition();
+
+    private MailSessionDefinition() {
+        super(new Parameters(MailExtension.MAIL_SESSION_PATH,
+                MailExtension.getResourceDescriptionResolver(MailSubsystemModel.MAIL_SESSION))
+                .setAddHandler(MailSessionAdd.INSTANCE)
+                .setRemoveHandler(new ServiceRemoveStepHandler(MailSessionDefinition.SESSION_CAPABILITY.getCapabilityServiceName(), MailSessionAdd.INSTANCE))
+                .setCapabilities(SESSION_CAPABILITY)
+        );
+        ApplicationTypeConfig atc = new ApplicationTypeConfig(MailExtension.SUBSYSTEM_NAME, MailSubsystemModel.MAIL_SESSION);
+        accessConstraints = new ApplicationTypeAccessConstraintDefinition(atc).wrapAsList();
```

Comment:
Parameters lets you set the access constraints, and then SimpleResourceDefinition deals with the registration.